accesses/services/accesses.py

```diff
@@ -477,4 +484,133 @@ def close_accesses(perimeters_to_delete: QuerySet):
+        For later syncing, inactive Practitioners will have their corresponding profiles deactivated.
+        @return: None
+        """
+        practitioner_res = self.fhir_client.resources("Practitioner")
+
+        for p in practitioner_res.fetch_all():
+            if p.active:
+                practitioner = FhirPractitioner(username=p.identifier,
+                                                firstname=p.name.official.given,
+                                                lastname=p.name.family,
+                                                email=p.telecom.value)
+                user, created = User.objects.get_or_create(defaults=practitioner.__dict__,
+                                                           username=practitioner.username)
+                if created or not user.profiles.filter(source=ORBIS, is_active=True).exists():
+                    Profile.objects.create(user=user, source=ORBIS, is_active=True)
+            else:
+                linked_profiles = Profile.objects.filter(user_id=p.identifier, source=ORBIS, is_active=True)
+                linked_profiles.accesses.update(end_datetime=timezone.now())
+                count_deactivated, _ = linked_profiles.update(is_active=False)
+                _logger.info(f"{count_deactivated} ORBIS profiles have been deactivated for user `{p.identifier}`")
+        _logger.info(f"{practitioner_res.count()} users have been synchronized with ORBIS")
+
+
+    def sync_accesses(self):
+        """
+        - Fetch the PractitionerRole resource which represents an authorization for a Practitioner
+          to perform a specific Role for a give period.
+        - The roles fetched from FHIR are mapped to predefined roles in the DB.
+        - For active PractitionerRoles, ensure having the corresponding trio (role, profile, perimeter)
+          is available before creating a new access.
+        - For later syncing, if a PractitionerRole gets deactivated, the corresponding Access is deactivated.
+        cf. https://hl7.org/fhir/R4/practitionerrole.html
+        @return: None
+        """
+        practitioner_role_res = self.fhir_client.resources("PractitionerRole")
+        skipped_roles = []
+        count_new_accesses = 0
+        for pr in practitioner_role_res.fetch_all():
+            if pr.active:
+                fpr = FhirPractitionerRole(id=pr.identifier,
+                                           active=pr.active,
+                                           practitioner_id=pr.practitioner,
+                                           perimeter_id=int(pr.organization),
+                                           role_name=pr.code,
+                                           start_datetime=pr.period.start,
+                                           end_datetime=pr.period.end)
+                role = self.get_mapped_role(fpr.role_name)
+                profile = Profile.objects.filter(user_id=fpr.practitioner_id).first()
```

Comment:
the profile may not exist, so we have to skip processing this practitioner role.
Alternatively, we could call (in batch) something like :
```
self.fhir_client.resource(
        'PractitionerRole',
        practitioner=[user_id for user_id in Profile.objects.filter(source="Orbis").all()],
        active=True
    )
```
to get only the PractitionerRole of interest